applications/asset_tracker_v2/prj_nrf9160dk_nrf9160ns.conf

```diff
@@ -0,0 +1,106 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# General config
+CONFIG_NEWLIB_LIBC=y
+CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=y
+CONFIG_ASSERT=y
+CONFIG_REBOOT=y
+CONFIG_RESET_ON_FATAL_ERROR=n
+
+# Heap and stacks
+CONFIG_HEAP_MEM_POOL_SIZE=32768
+CONFIG_MAIN_STACK_SIZE=1024
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+CONFIG_HW_STACK_PROTECTION=y
+
+# Logging
+CONFIG_LOG=y
+CONFIG_LOG_IMMEDIATE=y
+
+# DK - Used for buttons and LEDs in UI module.
+CONFIG_DK_LIBRARY_INVERT_LEDS=n
+
+# BSD library
+CONFIG_BSD_LIBRARY=y
+
+# Network
+CONFIG_NETWORKING=y
+CONFIG_NET_NATIVE=n
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_OFFLOAD=y
+
+# LTE link control
+CONFIG_LTE_AUTO_INIT_AND_CONNECT=n
+CONFIG_LTE_NETWORK_MODE_LTE_M_GPS=y
+## Power saving timers.
+CONFIG_LTE_PSM_REQ_RPTAU="01000011"
+CONFIG_LTE_PSM_REQ_RAT="00000000"
+
+# AWS IoT
+## Mandatory application specific configurations for AWS_IOT.
+CONFIG_AWS_IOT=y
+CONFIG_AWS_IOT_TOPIC_UPDATE_DELTA_SUBSCRIBE=y
+CONFIG_AWS_IOT_AUTO_DEVICE_SHADOW_REQUEST=y
+CONFIG_AWS_IOT_MQTT_RX_TX_BUFFER_LEN=2048
+CONFIG_AWS_IOT_MQTT_PAYLOAD_BUFFER_LEN=2048
+CONFIG_AWS_IOT_APP_SUBSCRIPTION_LIST_COUNT=1
+CONFIG_AWS_IOT_CLIENT_ID_APP=y
+CONFIG_AWS_FOTA=y
+
+## AWS IoT options that must be configured.
+CONFIG_AWS_IOT_SEC_TAG=42
+CONFIG_AWS_IOT_BROKER_HOST_NAME="example-hostname.aws.com"
+
+# MQTT
+# Keepalive should be set to the maximum specified MQTT keepalive timeout of the
+#        selected cloud library.
+#        1200 seconds for AWS IoT.
+CONFIG_MQTT_KEEPALIVE=1200
+
+# Settings - Used to store real-time device configuration to flash.
+CONFIG_SETTINGS=y
+CONFIG_SETTINGS_FCB=y
+CONFIG_FCB=y
+
+# FOTA - Needed by AWS IoT.
+CONFIG_FOTA_DOWNLOAD=y
+CONFIG_DFU_TARGET=y
+CONFIG_DOWNLOAD_CLIENT=y
+CONFIG_DOWNLOAD_CLIENT_HTTP_FRAG_SIZE_1024=y
+CONFIG_DOWNLOAD_CLIENT_STACK_SIZE=4096
+CONFIG_DOWNLOAD_CLIENT_BUF_SIZE=2300
```

Comment:
```suggestion
CONFIG_DOWNLOAD_CLIENT_BUF_SIZE=2300
CONFIG_DOWNLOAD_CLIENT_MAX_HOSTNAME_SIZE=128
CONFIG_DOWNLOAD_CLIENT_MAX_FILENAME_SIZE=64
```

Reply:
The current defaults do not work with hostnames and filenames generated by the Asset Tracker Cloud Example, therefore we need to provide safe defaults here.

Reply:
Added the hostname size config, but keeping the default filename size as it is larger by default (192) and it seems users tend to use large filenames (#3774)